precompiles/erc20/integration_test.go

```diff
@@ -422,6 +427,8 @@ var _ = Describe("ERC20 Extension -", func() {
 					)
 
 					// Check that the allowance was removed since we authorized only the transferred amount
+					// FIXME: This is not working for the case where we transfer from the own account
+					// because the allowance is not removed on the SDK side.
 					is.ExpectNoSendAuthzForContract(
 						callType, contractsData,
 						owner.Addr, owner.Addr,
```

Comment:
The case where we `transferFrom` the owner address will not be possible to fully support without adjusting our Cosmos SDK fork, since in Cosmos it's not supported to create authorizations for the same account. This is possible in ERC20s though.

However, the current alignment leads to the shown problem in this test - once set up through the EVM extension, the allowance is not reduced because in `tx.go` we use `p.AuthzKeeper.DispatchActions(...)`. In that function, the case where granter == grantee is being skipped:

https://github.com/evmos/cosmos-sdk/blob/v0.47.5-evmos/x/authz/keeper/keeper.go#L97-L133